certgen/certProcessor/src/main/java/org/incredible/certProcessor/views/PdfGenerator.java

```diff
@@ -0,0 +1,94 @@
+package org.incredible.certProcessor.views;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.incredible.certProcessor.JsonKey;
+import org.incredible.pojos.CertificateExtension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import static org.apache.commons.lang.StringUtils.capitalize;
+
+public class PdfGenerator {
+
+    private static Logger logger = LoggerFactory.getLogger(PdfGenerator.class);
+    private static ObjectMapper mapper = new ObjectMapper();
+    private static final String PRINT_SERVICE_URL = "print-service:5000/v1/print/pdf";
+
+    public static String generate(String htmlTemplateUrl, CertificateExtension certificateExtension , Map<String,Object> qrMap,
```

Comment:
is it just a QR Map? We need to give all that HTMLVars into the print-service. Rename if its name problem.

Reply:
Its just QR related data like access code , QR image link , and its file path.

Reply:
I will still insist, this must be named details - map. Variable names live long and must be carefully thought through. This is the payload to print-service and could be extended, which we don't know.

I think htmlvars or kvPair (keyvalues) as generic could make sense.

If you call a map to say 'qrMap', the tendency for the next developer will be to create another param for something else. We must not lead to it.